Python/structmember.c

```diff
@@ -281,8 +306,14 @@ PyMember_SetOne(char *addr, PyMemberDef *l, PyObject *v)
         break;
     case _Py_T_OBJECT:
     case Py_T_OBJECT_EX:
-        oldv = *(PyObject **)addr;
-        *(PyObject **)addr = Py_XNewRef(v);
+#ifdef Py_GIL_DISABLED
+        Py_BEGIN_CRITICAL_SECTION(obj);
+#endif
+        oldv = FT_ATOMIC_LOAD_PTR(*(PyObject **)addr);
```

Comment:
This can just be `oldv = *(PyObject **)addr;` because it's reading within the lock.